### PR TITLE
Add SAFETY comments presentation

### LINKED ISSUE #20: Document `// SAFETY:` comments.
We should document what we expect from `// Safety: ..` comments.

See also https://github.com/rust-lang/rust/issues/66219

### PR DESCRIPTION
**Pull request by poliorcetics**
Fix #20.

### CODE REVIEW

File: src/documentation/safety-comments.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Safety comments
+
+Using [`unsafe`] blocks in often required in the Rust compiler or standard
```
Comment (JohnTitor) [inline src/documentation/safety-comments.md:3]:
```suggestion
Using [`unsafe`] blocks is often required in the Rust compiler or standard
```

File: src/documentation/safety-comments.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+[`Send`]: https://doc.rust-lang.org/stable/std/marker/trait.Send.html
+[`Sync`]: https://doc.rust-lang.org/stable/std/marker/trait.Sync.html
+[as_bytes_mut]: https://github.com/rust-lang/rust/blob/a08f25a7ef2800af5525762e981c24d96c14febe/library/core/src/str/mod.rs#L278
+[`unsafe_op_in_unsafe_fn`]: https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unsafe-op-in-unsafe-fn
+
+## Inside *safe* elements
+
+Inside safe elements, a `SAFETY:` comment must not depend on anything from the
+caller beside properly contruscted types and values (i.e, if your function
```
Comment (JohnTitor) [inline src/documentation/safety-comments.md:57]:
```suggestion
caller beside properly constructed types and values (i.e, if your function
```

File: src/documentation/safety-comments.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+receive a reference that is unaligned or null, it's the caller fault if it fails
```
Comment (JohnTitor) [inline src/documentation/safety-comments.md:58]:
```suggestion
receives a reference that is unaligned or null, it's the caller fault if it fails
```

File: src/documentation/safety-comments.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+library:
+
+[`unsafe`]: https://doc.rust-lang.org/stable/std/keyword.unsafe.html
+
+## Inside `unsafe` elements
+
+This one shows how an `unsafe` function can pass the requirements through to its
+caller with the use of documentation in a `# Safety` section while still having
+more invariants needed that are not required from callers. `clippy` has a
+lint for `# Safety` sections by the way.
```
Comment (JohnTitor) [inline src/documentation/safety-comments.md:16]: This is just my favor but placing a link above the example is easier to find.

```suggestion
lint for `# Safety` sections by the way ([see the function on GitHub][as_bytes_mut]):
```

**Comment by poliorcetics** [comment]
What's the status on this ?

**Comment by yaahc** [comment]
> What's the status on this ?

Looks like it's good to go! Thank you again for writing this up and I apologize for the wait.

**Pull request merged**